src/server.jl

```diff
@@ -29,6 +29,38 @@ end
 ##############################
 # server parameter conversions
 ##############################
+struct Param
+    keylist::Vector{String}
+    value::String
+end
+
+function parse_query_dict(query_dict::Dict{String, String})::Vector{Param}
+    params = Vector{Param}()
+    for (key, value) in query_dict
+        keylist = replace.(split(key, "["), "]"=>"")
+        push!(params, Param(keylist, value))
+    end
+
+    return params
+end
+
+function convert_to_dict(params::Vector{Param})::Dict{String, Any}
+    deserialized_dict = Dict{String, Any}()
+
+    for param in params
+        current = deserialized_dict
+        for part in param.keylist[1:end-1]
+            current = get!(current, part, Dict{String, Any}())
+        end
+        current[param.keylist[end]] = param.value
+    end
+    return deserialized_dict
+end
+
+function deep_dict_repr(qp::Dict)
+    convert_to_dict(parse_query_dict(qp))
```

Comment:
I don't see `convert_to_dict` used anywhere else. Why not rename `convert_to_dict` to `deep_dict_repr`?

Reply:
Yes will remove, residue of refactoring